api/apps/articles/routes.coffee

```diff
@@ -54,7 +54,7 @@ User = require '../users/model.coffee'
 
   Article.destroy req.article._id, (err) ->
     return next err if err
-    res.send present req.article
+    present(req.article).then((results) => res.send(results))
```

Comment:
This was kind of an annoying result of `await`-ing on the db calls, but this works -- do you foresee any issues with this?

Reply:
Yeah I realized that this would change the signature to an async/promise-returning function, resulting in trickling refactors 😞  For a fun related read about this challenge with most languages check out [this article](http://journal.stuffwithstuff.com/2015/02/01/what-color-is-your-function/).